### DISCUSSION TITLE
Record page load and then redirect immediately?

### DISCUSSION THREAD

**Discussion by zenblender** [plausible/analytics#1274] [Self-Hosted Support]
I wanted to use Plausible to track incoming newsletter clicks where the URL contains UTM fields such as `utm_source` (from https://plausible.io/blog/utm-tracking-tags). However, I don't want to have these fields persist in the URL. What I want to do is have a simple page that tracks the page with its UTM fields in Plausible, then redirect to my home page with a "clean" URL.
I have something roughly like this:
```
<html>
  <head>
    <script async defer data-domain="example.com" src="[path-to-my-plausible-js-file]"></script>
    <script>
      window.location = '/';
    </script>  
  </head>
  <body>
    <h3>Redirecting...</h3>
  </body>
</html>
```
As written, I wouldn't expect this to work. I assume that `window.location = '/';` would happen before the page is tracked by Plausible. What I really want is to ensure that Plausible tracks the page (and its UTM fields) _before_ the redirect happens.

Is this possible? I can of course add an arbitrary delay before redirecting, and maybe altering the `async` and `defer` attributes would improve the likelihood of being tracked, but this seems like a timing issue. I just wanted an approach that is reliably going to wait for the Plausible things to finish before redirecting. Thanks for any help!

**Comment by ukutaht** [comment]
Hey. That's a good question and ideally we would provide a clean way to accomplish this. Unfortunately we don't offer an easy way to do this at the moment so a delay will be the best bet.
